### PR TITLE
fix: send ETH_P_ALL in htons format

### PR DESCRIPTION
**Pull request by tzneal**

### CODE REVIEW

**Comment by tzneal** [comment]
This was required in my testing on Linux at least.   I *think* the other values need the same treatment, but didn't want to change them as I didn't test on the other platforms.

File: src/sys/socket/mod.rs [hunk 1]
```diff
@@ -217,7 +224,7 @@ pub enum SockProtocol {
     // The protocol number is fed into the socket syscall in network byte order.
     #[cfg(any(target_os = "android", target_os = "linux"))]
     #[cfg_attr(docsrs, doc(cfg(all())))]
-    EthAll = libc::ETH_P_ALL.to_be(),
+    EthAll = htons_i32(libc::ETH_P_ALL),
```
Comment (asomers) [inline src/sys/socket/mod.rs:227]: Would this work instead?  It's a little bit simpler to inline it.
```suggestion
    EthAll = (libc::ETH_P_ALL as u16).to_be() as i32,
```

Comment (tzneal) [reply src/sys/socket/mod.rs:227]: I think it works the same, but it's a `const fn` so it's evaluated at compile time and won't require inlining, or did you just mean it's simpler to not have the function?

Happy to do it either way, but I *think* the other values need the same treatment on Linux, I just hadn't tested that far.

Comment (asomers) [reply src/sys/socket/mod.rs:227]: Oh!  It would really help if you could test them too.  I suggested inlining that function just for readability's sake.

**Comment by collinoc** [comment]
Any status updates on this? Attaching [this relevant Rust forums post](https://users.rust-lang.org/t/nix-unexpected-behavior-from-af-packet-socket/89761) that looks into some of the issues occurring from this.

**Comment by tzneal** [comment]
From what I can tell, just the other `ETH_` constants need this handling.  The only one in nix at the moment is `ETH_P_ALL`, so I think it's good.

**Comment by asomers** [comment]
@tzneal are you going to inline that function like I suggested?  And this needs a CHANGELOG, too.

**Comment by tzneal** [comment]
Yeah I can, I was going to leave it for the next `ETH_` case, but inlining is fine with me.

**Review by asomers — APPROVED** [review]
bors r+

**Pull request merged**